dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -418,6 +497,16 @@ class CZipExtract :
 
             do
             {
+                if (*bCancel)
+                {
+                    CloseHandle(hFile);
+                    BOOL deleteResult = DeleteFileA(FullPath);
+                    if(deleteResult != 0)
```

Comment:
`DeleteFile` returns zero on failure, so you'll want to check `== 0` here (and print `GetLastError()` below rather than the result)